erigon-lib/diagnostics/network.go

```diff
@@ -2,60 +2,99 @@ package diagnostics
 
 import (
 	"context"
+	"sort"
+	"sync"
+	"time"
 
 	"github.com/ledgerwatch/log/v3"
 )
 
-func (d *DiagnosticClient) setupNetworkDiagnostics(rootCtx context.Context) {
-	d.runCollectPeersStatistics(rootCtx)
+type PeerStats struct {
+	peersInfo     sync.Map
```

Comment:
i think must be *sync.Map

Reply:
Why ? I don't think that will make any difference.

Reply:
I don't know how to right use sync.Map - I mean https://go.dev/tour/methods/8

Reply:
Done